### PR TITLE
Flask login

### PR DESCRIPTION
**Pull request by zfrank7777**

### CODE REVIEW

**Comment by zfrank7777** [comment]
The only thing undone is that the response of an unauthorized access to login_required API.
For now the only login_required API is delete_user.

File: src/surfing_penguin/routes.py [hunk 1]
```diff
@@ -120,51 +121,11 @@ def post(self):
...
-        {
-            'question': 'test Q2'
-        }
-    ]
-    return render_template('filling_in.html', qstnr=qstnr)
+@login_manager.unauthorized_handler
+@api.marshal_with(api_return_message)
+def unauthorized():
+    print("entering")
+    return {'message': "Please Login First"}
```
Comment (amjltc295) [inline src/surfing_penguin/routes.py]: Please add a \# TODO tag if it's not done yet.

File: src/surfing_penguin/__init__.py [hunk 2]
```diff
@@ -11,6 +12,9 @@
 surfing_penguin = Flask(__name__)
 surfing_penguin.config.from_object(Config)
 db_engine = create_engine('sqlite:///:memory:', echo=False)
+login_manager = LoginManager(surfing_penguin)
+login_manager.login_view = 'login'
+login_manager.init_app(surfing_penguin)
 
 api = Api(surfing_penguin)
```
Comment (amjltc295) [inline src/surfing_penguin/__init__.py:20]: Please add a \# NOQA in the end of this file

File: src/surfing_penguin/routes.py [hunk 3]
```diff
@@ -1,68 +1,54 @@
...
+api_show_user_and_time = api.model("show_user_and_time_model", {
+        'username': fields.String,
+        'last_seen': fields.DateTime
+    })
 """ user account associated APIs:
     register, login, show_users, delete_user, search_user """
 
 
+@surfing_penguin.before_request
+def before_request():
```
Comment (amjltc295) [inline src/surfing_penguin/routes.py:38]: It would be better to add comments for this function (and other not self-explained ones).

**Review by amjltc295 — CHANGES_REQUESTED** [review]
Overall looks good. Just some minor parts to be modified.

**Pull request merged**